lib/rubocop/cop/rspec/no_expectation_example.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks if an example includes any expectation.
+      #
+      # All RSpec's example and expectation methods are covered by default.
+      # If you are using your own custom methods,
+      # add the following configuration:
+      #
+      #   RSpec/NoExpectationExample:
+      #     ExampleMethodnames:
+      #       - custom_it
+      #     ExpectationMethodNames:
+      #       - custom_expect
+      #
+      # @example
+      #
+      #   # bad
+      #   it do
+      #     a?
+      #   end
+      #
+      #   # good
+      #   it do
+      #     expect(a?).to be_truthy
+      #   end
+      class NoExpectationExample < Base
+        extend AutoCorrector
+
+        include RangeHelp
+
+        DEFAULT_EXAMPLE_METHOD_NAMES = ::Set[
+          :example,
+          :fexample,
+          :fit,
+          :focus,
+          :fscenario,
+          :fspecify,
+          :it,
+          :its,
+          :pending,
+          :scenario,
+          :skip,
+          :specify,
+          :xexample,
+          :xit,
+          :xscenario,
+          :xspecify,
+        ]
+
+        DEFAULT_EXPECTATION_METHOD_NAMES = ::Set[
+          :expect_any_instance_of,
+          :expect,
+          :is_expected,
+          :should_not_receive,
+          :should_not,
+          :should_receive,
+          :should,
+        ]
+
+        MSG = 'No expectation found in this example.'
+
+        # @param [RuboCop::AST::Node] node
+        def on_send(node)
+          return unless example_method_names.include?(node.method_name)
+
+          block = argument_block_of(node)
+          return unless block
+          return if including_any_expectation?(block)
+
+          add_offense(block) do |corrector|
+            corrector.remove(removed_range(block))
+          end
+        end
+
+        private
+
+        # @param [RuboCop::AST::Node] node
+        # @return [RuboCop::AST::BlcokNode, nil]
+        def argument_block_of(node)
+          block = node.parent
+          if block.is_a?(::RuboCop::AST::BlockNode) && block.send_node == node
+            block
+          end
+        end
+
+        # @return [Set<Symbol>]
+        def example_method_names
+          DEFAULT_EXAMPLE_METHOD_NAMES +
+            example_method_names_from_config
+        end
+
+        # @return [Set<Symbol>]
+        def example_method_names_from_config
+          cop_config.fetch('ExampleMethodNames', []).map(&:to_sym).to_set
+        end
+
+        # @param [RuboCop::AST::Node] node
+        # @return [Boolean]
+        def expectation_method_call?(node)
+          return false unless node.is_a?(::RuboCop::AST::SendNode)
+
+          expectation_method_names.include?(node.method_name)
+        end
```

Comment:
I've implemented it so that only configured methods are considered as expectation, but it might be better to put in a mechanism to judge like `method_name.start_with?('expect')`.

This would create a naming convention for the `expect` wrapper methods as discussed at https://github.com/rubocop/rubocop-rspec/issues/208#issuecomment-497893002.

> There are cases when this will result in false positives:
> 
> ```
> def expect_processed_value
>   expect(process(value))
> end
> 
> def surprise_the_crowd
>   eq(100)
> end
> 
> it 'is hard to detect an expectation here' do
>   expect_processed_value.to surprise_the_crowd
> end
> ```

Reply:
Note that rubocop-rspec already provides a way to add methods to be considered as expectation, by listing them under Language/Expectations config